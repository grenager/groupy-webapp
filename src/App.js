// import React from 'react';
// import logo from './logo.svg';
// import './App.css';
//
// function App() {
//   return (
//     <div className="App">
//       <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />
//         <p>
//           Edit <code>src/App.js</code> and save to reload.
//         </p>
//         <a
//           className="App-link"
//           href="https://reactjs.org"
//           target="_blank"
//           rel="noopener noreferrer"
//         >
//           Learn React
//         </a>
//       </header>
//     </div>
//   );
// }
//
// export default App;

import React, { Component } from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { ThemeProvider } from '@material-ui/styles';
import theme from './theme';
import Routes from './Routes';

const browserHistory = createBrowserHistory();

export default class App extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <Router history={browserHistory}>
          <Routes />
        </Router>
      </ThemeProvider>
    );
  }
}
