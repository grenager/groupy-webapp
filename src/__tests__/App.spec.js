import localStorage from 'localStorage';
import localStorageObject from './localStorageObject';
import App from '../App';

// localStorage is sync
const gatherLocalStorage = () => {
  return JSON.parse(localStorage.getItem('groupy'));
}

test('proper profile data is correct', async () => {
  const ls = await gatherLocalStorage();
  const userID = ls.userID;
  expect(ls).toMatchSnapshot();
  expect(userID).toEqual('5e62ade15014b544a50af943');
});

test('localStorageObject should have properties token and userID', () => {
  expect(localStorageObject).toHaveProperty('token');
  expect(localStorageObject).toHaveProperty('userID');
});
