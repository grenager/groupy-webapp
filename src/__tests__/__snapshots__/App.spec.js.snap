// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proper profile data is correct 1`] = `Promise {}`;
